packages/core/src/parameters/getParameterDefinitionsFromSource.js

```diff
@@ -0,0 +1,174 @@
+const JSON5 = require('json5')
+
+const countSpaces = (l) => {
+  let count = 0
+  for (let i = 0; i < l.length; i++) {
+    if (l[i] === ' ') count++
+    else if (l[i] === '\t') count += 2
+    else break
+  }
+  return count
+}
+
+const getParameterDefinitionsFromSource = (script) => {
+  const lines = []
+  script.split('\n').forEach((l, i) => {
+    const trim = l.trim()
+    if (trim) {
+      lines.push({ code: trim, line: l, lineNum: i + 1, indent: countSpaces(l) })
+    }
+  })
+
+  let i = 0; let lineNum; let code; let prev; let prevIndent
+  while (i < lines.length) {
+    code = lines[i].code
+    i++
+    if (code.length > 12 && code.indexOf('@jscad-params') !== -1) break
+  }
+
+  let groupIndex = 1
+  const defs = []
+
+  while (i < lines.length) {
+    code = lines[i].code
+    lineNum = lines[i].lineNum
+    if (code[0] === '}') break
+
+    const isGroup = code[0] === '/'
+    if (isGroup && prev) {
+      const isHint = prev.type === 'group' || prevIndent + prev.name.length <= lines[i].indent
+      if (isHint) {
+        prev.hint = prev.hint ? prev.hint + '\n' : ''
+        prev.hint += extractTextFromComment(code)
+        i++
+        continue
+      }
+    }
+
+    prevIndent = lines[i].indent
+    if (isGroup) {
+      // group
+      const name = '_group_' + (groupIndex++)
+      const def = parseComment(code, lineNum, name)
+      let caption = def.caption
+      if (caption[0] === '>') {
+        caption = caption.substring(1).trim()
+        if (!def.options) def.options = {}
+        def.options.initial = 'closed'
+      }
+
+      defs.push(prev = { name, type: 'group', caption, ...def.options })
+    } else {
+      const idx = code.indexOf('/')
+      if (idx === -1) {
+        const def = parseDef(code, lineNum)
+        def.caption = def.name
+        defs.push(prev = def)
+      } else {
+        defs.push(prev = parseOne(
+          code.substring(idx).trim(),
+          code.substring(0, idx).trim(),
+          lineNum, lineNum
+        ))
+      }
+    }
+    i++
+  }
+
+  return defs
+}
+
+const parseOne = (comment, code, line1, line2) => {
+  let def = parseDef(code, line2)
+  const { caption, options } = parseComment(comment, line1, def.name)
+  def.caption = caption || def.name
+  if (options) {
+    def = { ...def, ...options }
+    if (def.type === 'checkbox' && 'initial' in def) def.checked = true
+    if (def.type === 'slider') {
+      if (def.min === undefined) {
+        def.min = 0
+      }
+      if (def.max === undefined) {
+        def.max = 100
+      }
+    }
+  }
+
+  return def
+}
+
+const extractTextFromComment = (c) => {
+  const prefix = c.substring(0, 2)
+  if (prefix === '//') c = c.substring(2)
+  if (prefix === '/*') c = c.substring(2, c.length - 2)
```

Comment:
This is assuming that the comment is like `/* ... */`, so why not check for that?

Reply:
I think I need a more proper handling of multiline comments, maybe leave `/* ...*/` to be handled same as `//` with aditional check and thow an error saying multiline comments not supported by parameters parser ?

or some  other ide what to do with `/* ... */` when single line or multiline ?

Reply:
added better handling, and disallowed multiline comments. singliline `/* ... */` are allowed and work.